src/main/thrift/parquet.thrift

```diff
@@ -446,16 +446,19 @@ struct PageHeader {
   /** Compressed page size in bytes (not including this header) **/
   3: required i32 compressed_page_size
 
+  /** Encrypted page size in bytes (not including this header) **/
+  4: optional i32 encrypted_page_size
```

Comment:
I think you need to add this field at the end to be backwards compatible.

Reply:
1. I encrypted the compressed page using a fixed encryption algorithm(for the time being) based on an opinionated way like column starting with "encrypted_" will be encrypted in this way.
(I also think that the encryption status should come in the schema itself, but I was unable to add a field into the schema because of the lack of knowledge in that project.)
2. Since our page header is still not encrypted I think we can have the predicate push down(correct me if not). 
3. Since we are encrypting the compressed page I think it will act independently.
4. Currently I'm not specifying the algorithm details in the meta. If we can provide the same through the schema do we need it in the meta ?

Reply:
did you mean the following,

    /** Encrypted page size in bytes (not including this header) **/
  9: optional i32 encrypted_page_size

The encrypted_page_size  as the last field in the PageHeader.